src/iceberg/schema.h

```diff
@@ -78,6 +79,12 @@ class ICEBERG_EXPORT Schema : public StructType {
   Result<std::optional<std::reference_wrapper<const SchemaField>>> FindFieldById(
       int32_t field_id) const;
 
+  /// \brief Returns the full column name for the given id.
+  ///
+  /// \param field_id The id of the field to get the full name for.
+  /// \return The full name of the field with the given id, or std::nullopt if not found.
+  Result<std::optional<std::string>> FindColumnNameById(int32_t field_id) const;
```

Comment:
```suggestion
  Result<std::optional<std::string_view>> FindColumnNameById(int32_t field_id) const;
```

Use `std::string_view` to avoid a string copy.